pkg/controller/installation/products/codeready/reconciler.go

```diff
@@ -0,0 +1,381 @@
+	serverClient, err := pkgclient.New(r.restConfig, pkgclient.Options{})
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "could not create server client")
+	}
+
+	kcCfg, err := r.ConfigManager.ReadRHSSO()
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "could not retrieve sso config")
+	}
+	if kcCfg.GetRealm() == "" {
+		return v1alpha1.PhaseFailed, errors.New("keycloak realm is not set in sso config")
+	}
+	if kcCfg.GetURL() == "" {
+		return v1alpha1.PhaseFailed, errors.New("keycloak url is not set in sso config")
+	}
+	kcRealm := &keycloakv1.KeycloakRealm{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kcCfg.GetRealm(),
+			Namespace: kcCfg.GetNamespace(),
+		},
+	}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: kcCfg.GetRealm(), Namespace: kcCfg.GetNamespace()}, kcRealm)
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "could not retrieve keycloakrealm custom resource")
+	}
+
+	cheCluster := &chev1.CheCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultCheClusterName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: defaultCheClusterName, Namespace: r.Config.GetNamespace()}, cheCluster)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			if err = r.createCheCluster(ctx, serverClient, kcCfg, kcRealm); err != nil {
+				if k8serr.IsAlreadyExists(err) {
+					return v1alpha1.PhaseInProgress, nil
+				}
+				return v1alpha1.PhaseFailed, pkgerr.Wrap(err, fmt.Sprintf("could not create checluster custom resource in namespace: %s", r.Config.GetNamespace()))
+			}
+			return v1alpha1.PhaseInProgress, nil
+		}
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, fmt.Sprintf("could not retrieve checluster custom resource in namespace: %s", r.Config.GetNamespace()))
+	}
+	if cheCluster.Spec.Auth.ExternalKeycloak && !cheCluster.Spec.Auth.OpenShiftOauth && cheCluster.Spec.Auth.KeycloakURL == kcCfg.GetURL() && cheCluster.Spec.Auth.KeycloakRealm == kcCfg.GetRealm() && cheCluster.Spec.Auth.KeycloakClientId == defaultClientName {
+		return v1alpha1.PhaseInProgress, nil
+	}
+	cheCluster.Spec.Auth.ExternalKeycloak = true
+	cheCluster.Spec.Auth.OpenShiftOauth = false
+	cheCluster.Spec.Auth.KeycloakURL = kcCfg.GetURL()
+	cheCluster.Spec.Auth.KeycloakRealm = kcRealm.Name
+	cheCluster.Spec.Auth.KeycloakClientId = defaultClientName
+	if err = serverClient.Update(ctx, cheCluster); err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, fmt.Sprintf("could not update checluster custom resource in namespace: %s", r.Config.GetNamespace()))
+	}
+	return v1alpha1.PhaseInProgress, nil
+}
+
+func (r *Reconciler) handleAwaitingOperator(ctx context.Context) (v1alpha1.StatusPhase, error) {
+	logrus.Infof("checking installplan is created for subscription %s in namespace: %s", defaultSubscriptionName, r.Config.GetNamespace())
+	ip, err := r.mpm.GetSubscriptionInstallPlan("codeready-workspaces", r.Config.GetNamespace())
```

Comment:
Should that be a var?
```suggestion
	ip, err := r.mpm.GetSubscriptionInstallPlan(defaultSubscriptionName, r.Config.GetNamespace())
```